src/java/org/apache/cassandra/service/accord/AccordMessageSink.java

```diff
@@ -227,12 +225,33 @@ public void send(Node.Id to, Request request)
         messaging.send(message, endpoint);
     }
 
+    private static boolean isRangeBarrier(Request request)
+    {
+        if (!(request instanceof TxnRequest))
+            return false;
+
+        TxnRequest<?> txnRequest = (TxnRequest<?>) request;
+        if (!txnRequest.txnId.kind().isSyncPoint())
+            return false;
+
+        return txnRequest.txnId.domain().isRange();
+    }
+
     @Override
     public void send(Node.Id to, Request request, AgentExecutor executor, Callback callback)
     {
         Verb verb = getVerb(request);
         Preconditions.checkNotNull(verb, "Verb is null for type %s", request.type());
-        Message<Request> message = Message.out(verb, request);
+        Message<Request> message;
+        if (isRangeBarrier(request))
```

Comment:
Arguably is this problem specific to sync points or do all range transactions probably have this issue and should have the longer timeout?

Reply:
It's possible that all range txns have this issue. For the time being though, since range barriers do _and_ there's an existing timeout knob for them, I'm starting there. It's possible this may need to be extended for all offline range operations though.